package/suse-migration-services-spec-template

```diff
@@ -26,6 +26,7 @@ Group:            System/Management
 Source:           suse-migration-services.tar.gz
 BuildRoot:        %{_tmppath}/%{name}-%{version}-build
 Conflicts:        suse-migration-sle15-activation
+Conflicts:         suse-migration-pre-checks
```

Comment:
This cannot conflict. We will need to build both the pre-checks and the migration services into the Live ISO that is used to drive the migration as the migration services will need to run the pre-checks in `fix` mode, in the appropriate context, i.e. in the context of the system that is being upgraded and that is currently not running since we are running from within the ISO image.

Reply:
The "Conflicts" is not blocking the install of suse-migration-pre-check and suse-migration-services together on the iso as I have tested this and it works . Rather since suse-migration-pre-checks is installed in the same python lib dir as suse-migration-services, the build server is reporting a conflict and aborts. When I use the "Conflicts" it allows the SLE15-Migration rpm to build correctly.    I think Markus ran into this when the original pre-checks were added and that is why he had to do this do this PR:

https://github.com/SUSE/suse-migration-services/pull/210  

https://github.com/SUSE/suse-migration-services/pull/210/files#diff-0b55022eedc44d6b4db6f9cf8a542fbeb013f9572bbd0162fa4b03c222ba87f4R28


https://github.com/SUSE/suse-migration-services/pull/210/files#diff-b8f5e9f254ff34608785645332e830f97640e3c924f81f53ee8563fd31e3a660R29

Reply:
So the workaround proposed by Keith should work fine since the precedence has already been established.  Problem with all three templates sharing the same setup.py is that there will be conflicts amongst all of them. One alternative would be that isolated the shared code into a common package and have all three packages depended on it, as Keith proposed in the email thread. Otherwise, we can try to restructure setup.py and the directories to produce multiple packages.

Reply:
Based on conversations between @rtamalin and I , a new solution, leveraging one spec file with "sub pacakges" was implemented and tested.